### PR TITLE
Enhance testing: Skip fused_optimizer tests if not supported.

### PR DESCRIPTION
**Pull request by vshekhawat-hlab**
Enhance testing: Skip fused_optimizer tests if not supported.

Added condition check to skip fused_optimizer tests if FusedAdam and FusedLamb are not supported by the accelerator. This enhancement ensures that the tests are appropriately skipped when the hardware configuration does not support these optimizers, preventing potential issues.

Details:
- Introduced a condition check to determine support for FusedAdam and FusedLamb.
- If not supported, fused_optimizer tests are skipped to improve test reliability.
- Improved compatibility and stability across different hardware configurations.

### CODE REVIEW

**Comment by vshekhawat-hlab** [comment]
@microsoft-github-policy-service agree company="Intel Corporation"

**Review by tjruwase — APPROVED** [review]

**Review by loadams — APPROVED** [review]

File: tests/unit/elasticity/test_elastic.py [hunk 1]
```diff
@@ -150,6 +150,7 @@ def test_proper_mbsz(ds_config):
 class TestNonElasticBatchParams(DistributedTest):
     world_size = 2
 
+    @pytest.mark.skipif(not deepspeed.ops.__compatible_ops__[FusedLambBuilder.NAME], reason="lamb is not compatible")
```
Comment (loadams) [inline tests/unit/elasticity/test_elastic.py:153]: @vshekhawat-hlab - could you run the pre-commit checks on this, that will fix the formatting check on this PR.

**Comment by loadams** [comment]
@vshekhawat-hlab - could you resolve the merge conflicts on this PR?

File: tests/unit/elasticity/test_elastic.py (hunk 1 above)
Comment (loadams) [reply tests/unit/elasticity/test_elastic.py:153]: Thanks!

**Comment by vshekhawat-hlab** [comment]
> @vshekhawat-hlab - could you resolve the merge conflicts on this PR?

@loadams,

Can you please check it again.

**Pull request merged**